archive/archive_test.go

```diff
@@ -242,6 +242,28 @@ func TestArchive_dirWithSymlink(t *testing.T) {
 	}
 }
 
+func TestArchive_dirWithSymlinkToFile(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("git symlinks don't work on Windows")
+	}
+
+	path := filepath.Join(testFixture("archive-symlink-file"), "link")
+	r, err := CreateArchive(path, new(ArchiveOpts))
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	expected := []string{
+		"link/",
+		"link/foo.txt",
+	}
```

Comment:
Missing test files that added with b4f681e9b65b887d5a3f85adedbb2ac253af837c commit.

```diff
 expected := []string{
-        "link/",
-        "link/foo.txt",
+        "deeper/",
+        "deeper/adeeperlink",
+        "deeper/linklink",
+        "deeper/linklinklink",
+        "link",
 }
```